### PR TITLE
Login

### PR DESCRIPTION
**Pull request by lee-merrill**
Added a login and an "add user" page.

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -75,6 +77,17 @@ var kvsPath = './tmp/keyValStore_' + config.networkId;
...
+  fdata = fs.readFileSync(PasswdFile).toString();
+}
+var userList = fdata.trim().split('\n');
+userList.forEach(function(userLine) {
+  var wd = userLine.split(':');
+  appUsers.push({userName: wd[0], userPw: wd[1]});
+});
+
 //Start building the appUsers data structure
 config.users.forEach(function(name) {
```
Comment (ericmvaughn) [inline api.js]: Did you intend on keeping the users from the configuration file or replace them entirely from the users in the password file?

File: api.js [hunk 2]
```diff
@@ -75,6 +77,17 @@ var kvsPath = './tmp/keyValStore_' + config.networkId;
...
 
+var PasswdFile = './passwd';
+var fdata = '';
+if (fs.existsSync(PasswdFile)) {
+  fdata = fs.readFileSync(PasswdFile).toString();
+}
+var userList = fdata.trim().split('\n');
+userList.forEach(function(userLine) {
+  var wd = userLine.split(':');
+  appUsers.push({userName: wd[0], userPw: wd[1]});
```
Comment (ericmvaughn) [inline api.js]: Should we include the userType in this push to appUsers as well?

File: public/js/controller.js [hunk 3]
```diff
@@ -362,4 +362,58 @@ var activeUser;
       templateUrl: 'templates/eventList.html'
     };
   });
+
+  myApp.controller('loginCtrl', ['$scope', '$http',
+  function($scope, $http) {
+    var jsonStr = '{}';
```
Comment (ericmvaughn) [inline public/js/controller.js]: Is jsonStr needed for anything or can it be left off of the get call?

File: public/templates/addUser.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    <label for="passwordRepeat" class="col-sm-2 control-label">Repeat Password</label>
+    <div class="col-sm-8">
+      <input type="password" class="form-control" id="userPwRepeat" name="userPwRepeat" ng-model="userPwRepeat">
+    </div>
+  </div>
+  <div class="form-group">
+    <label for="userType" class="col-sm-2 control-label">User Type</label>
+    <div class="col-sm-8">
+      <select class="form-control" id="userType" ng-model="userType">
+        <option value="">User</option>
```
Comment (ericmvaughn) [inline public/templates/addUser.html]: Should this be value="user"?

File: api.js (hunk 1 above)
Comment (lee-merrill) [inline api.js]: I was going to replace them with the password file.

File: api.js (hunk 2 above)
Comment (lee-merrill) [inline api.js]: Yes, good point...

File: public/js/controller.js (hunk 3 above)
Comment (lee-merrill) [inline public/js/controller.js]: It's not needed, so I removed it.

File: public/templates/addUser.html (hunk 4 above)
Comment (lee-merrill) [inline public/templates/addUser.html]: This is the only way I could get a default of "User" to show in the control, so I just consider a null value to be "User".

**Pull request merged**